msal/application.py

```diff
@@ -250,10 +256,22 @@ def _build_client(self, client_credential, authority):
             assert ("private_key" in client_credential
                     and "thumbprint" in client_credential)
             headers = {}
+            unencrypted_private_key = client_credential['private_key']
             if 'public_certificate' in client_credential:
                 headers["x5c"] = extract_certs(client_credential['public_certificate'])
+            if 'password' in client_credential:
+                if isinstance(client_credential['private_key'], six.text_type):
+                    private_key = client_credential['private_key'].encode(encoding="utf-8")
+                else:
+                    private_key = client_credential['private_key']
+                if isinstance(client_credential['password'], six.text_type):
+                    password = client_credential['password'].encode(encoding="utf-8")
```

Comment:
That byte conversion is needed only because load_pem_private_key() which is the new introduction here accepts private key in bytes, I believe the Jwt assertion api we use - `jwt.encode` does not necessitate such a byte object for key.